src/test/java/org/opentripplanner/pricing/transit/TransitPriceCalculatorTest.java

```diff
@@ -0,0 +1,358 @@
+        |<--------->|<--->|<----------------------------------->|
+        0       7   10    15                        40     46   50    Arrive at stop time (minutes)
+        |-------|---|-----|-------------------------|------|----|
+        |           |     |                                     |
+        |   105     |walk |               13                    |     Mean of transport
+
+         */
+        tripStages.add(new TransitTripStage(firstRoute, stop1, 1, 0));
+        tripStages.add(new TransitTripStage(firstRoute, stop8, 8, 0));
+        tripStages.add(new TransitTripStage(firstRoute, stop11, 11, 0));
+        tripStages.add(new TransitTripStage(secondRoute, stop16, 16, 0));
+        tripStages.add(new TransitTripStage(secondRoute, stop41, 41, 0));
+        tripStages.add(new TransitTripStage(secondRoute, stop47, 47, 0));
+        tripStages.add(new TransitTripStage(secondRoute, stop51, 51, 0));
+
+        TransitTripDescription tripDescription = new TransitTripDescription(tripStages);
+
+        BigDecimal transitPrice = priceCalculator.computePrice(tripDescription);
+
+        assertEquals(0, transitPrice.compareTo(BigDecimal.valueOf(6.8)));
+    }
+
+    @Test
+    public void shouldReturn0PriceWhenNoTicketMatchesNoFare() {
+        priceCalculator.getAvailableTickets().add(zoneAOnlyTicket);
+
+        Route firstRoute = new Route();
+        firstRoute.setId(new FeedScopedId("ZTM", "105"));
+        firstRoute.setShortName("105");
+        Route secondRoute = new Route();
+        secondRoute.setId(new FeedScopedId("ZTM", "13"));
+        secondRoute.setShortName("13");
+
+        Stop stop1 = new Stop();
+        stop1.setZoneId("2");
+        stop1.setId(new FeedScopedId());
+        Stop stop8 = new Stop();
+        stop8.setZoneId("2");
+        stop8.setId(new FeedScopedId());
+        Stop stop11 = new Stop();
+        stop11.setZoneId("2");
+        stop11.setId(new FeedScopedId());
+        Stop stop16 = new Stop();
+        stop16.setZoneId("2");
+        stop16.setId(new FeedScopedId());
+        Stop stop41 = new Stop();
+        stop41.setZoneId("1/2");
+        stop41.setId(new FeedScopedId());
+        Stop stop47 = new Stop();
+        stop47.setZoneId("1");
+        stop47.setId(new FeedScopedId());
+        Stop stop51 = new Stop();
+        stop51.setZoneId("1");
+        stop51.setId(new FeedScopedId());
+
+        List<TransitTripStage> tripStages = new ArrayList<>();
+
+        /*
+
+        Routes used in the tested itinerary:
+
+        |  10 min   |5 min|             35 min                  |     Travel time
+        |<--------->|<--->|<----------------------------------->|
+        0       7   10    15                        40     46   50    Arrive at stop time (minutes)
+        |-------|---|-----|-------------------------|------|----|
+        |           |     |                                     |
+        |   105     |walk |               13                    |     Mean of transport
```

Comment:
+1 for those diagram.
It will be +2 if another row with ticket valid time.
But it is only nitpick and I think there are more important thinks to do :)